### ISSUE TITLE
issues running webprotege using docker image

### ISSUE THREAD

**Issue by niva83** [protegeproject/webprotege#692]
Hi there, 

I followed the instructions of running webprotege from a docker image.
However, there are a few issues.

1. Following errors are displayed while lunching the docker image:

```
webprotege_1  | 2020-06-21 07:15:45.501  INFO   Initializing WebProtege 
webprotege_1  | 2020-06-21 07:15:46.295  INFO   Loading webprotege.properties from the class path 
webprotege_1  | 2020-06-21 07:15:46.301  INFO   Overriding webprotege.mongodb.host with system property value: wpmongo 
webprotege_1  | 2020-06-21 07:15:46.316  INFO   No username has been supplied for the MongoDB client connection.  Authentication will not be used. 
webprotege_1  | 2020-06-21 07:15:46.324  INFO   Created MongoClient database connection without credentials for authentication 
webprotege_1  | 2020-06-21 07:15:46.856  INFO   Created MongoClient database connection 
```

and

```
webprotege_1  | 2020-06-21 07:16:08.452  WARN   System notification email address is not specified 
webprotege_1  | 2020-06-21 07:16:08.457  WARN   Application host is not specified 
webprotege_1  | 2020-06-21 07:16:08.458  WARN   Cannot construct well-formed application URL. (java.net.URISyntaxException: Expected hostname at index 8: https://:443) 
webprotege_1  | 2020-06-21 07:16:08.459  WARN   Cannot construct well-formed project URLs. (java.net.URISyntaxException: Expected hostname at index 8: https://:443#projects/3283960b-6a73-47c2-83b0-4943bb13d95d/edit/Classes) 
webprotege_1  | 2020-06-21 07:16:08.460  WARN   Cannot construct well-formed OWL Entity URLs. (java.net.URISyntaxException: Expected hostname at index 8: https://:443#projects/c7a3bc56-73aa-4cf4-abc0-aba3a63aa7d7/edit/Classes?selection=Class(owl:Thing)) 
```

2. When I fire up `http://localhost:5000/`, an error is diplayedin the GUI `WebProtégé is not configured properly`
3. Where and how to make user and admin accounts? Is there a default admin username/pass?
4. Listing the content of webprotege folder, folder `logsdir_IS_UNDEFINED` is present

Basically are there any additional steps for configuring webprotege once the docker image starts ?

**Comment by niva83** [comment]
@matthewhorridge it seems that there are additional steps once the docker image is running which are provided here:
https://github.com/protegeproject/webprotege/wiki/WebProt%C3%A9g%C3%A9-Deployment-via-Docker

I think it would be great to actually add the link to the above instructions in your README.MD under [Running from Docker](https://github.com/protegeproject/webprotege/wiki/WebProt%C3%A9g%C3%A9-Deployment-via-Docker section).

**Comment by katsi** [comment]
@niva83 +1
That part I had to dig out from somewhere else, so having it in README or at least a link to it would be great.

**Comment by niva83** [comment]
@katsi indeed, I've spent about 3+ hours trying to figure out why docker does not work as it should be. As you said, I was digging through the Git repo and by chance found the above description which helped me to make the docker ver of webprotege to work. Some people are not as pesistent in trying to dig out information, so indeed, either fusing the info from the link to README or at least puttin a link for **Additional steps** to make docker to work would be of great help for anyone new to webprotege and docker.

**Comment by katsi** [comment]
Has anyone here deployed webprotege with kubernetes? I am currently trying to do that and I have trouble connecting the frontend with mongo

**Comment by niva83** [comment]
@katsi I haven't, still working with dockers.

**Comment by katsi** [comment]
go it working, so if anyone needs help with that, am happy to share!

**Comment by niva83** [comment]
@katsi I think it would be helpful, I hope someone from the repo maintainers will catch up with the information we are sharing in this issue.

**Comment by mmanela** [comment]
@katsi  Can you share? I am hitting an issue where the web front-end cant find mongodb even though I can connect from the same container use mongoshell

**Comment by katsi** [comment]
I am happy to share. Wondering what format would make sense? Should we have
an informal webinar on the topic?

Katariina

**Comment by niva83** [comment]
@katsi webinar would be great. Maybe to record it and share over Youtube?

**Comment by katsi** [comment]
@mmanela how do you notice that the webprotege front-end cannot find the monodb? I am wondering, if afterall I am hitting the same problem. After configuring the admin account, I still get the notification that Webprotege is not configured properly AND the frontend misses some components that would otherwise be there if I would start it with docker-compose.

But what do work are my admin credentials.

**Comment by mmanela** [comment]
I was actually able to run the jar command to create the admin account (which was able to connect to the mongodb). But when I went to the website it gave me the mongodb connection error mentioned above. 

I was eventually able to resolve it by restarting the pods.

**Comment by katsi** [comment]
Two of my colleagues, one using Windows and the other Linux, reported a problem where the `webprotege.mongodb.host=wpmongo` was not stored when the contaienrs were launched. One solution here was to set the env with running the jar command, so like this: `env "webprotege.mongodb.host=wpmongo" java -jar /webprotege-cli.jar create-admin-account`

**Comment by katsi** [comment]
@niva83 What software could I use for that? Zoom, Google Meet, ..? Any recommendations? Never held a webinar before!

**Comment by niva83** [comment]
@katsi I've been using all above. Probably Google Meet is the easiest to setup if you already have Google account.

**Comment by niva83** [comment]
@johardi I noticed that you have updated readme.md with a better description on how to get webprotege to run from docker image. What is unclear is where and how webprotege.properties and mail.properties via docker image?

This question was originally posted as an issue #682 . 

As it is now, if the instructions are followed for running webprotege as a docker image, the problem is that we are unable to reset password or get email notification since mail.properties are not set.

**Comment by mikeandescavage** [comment]
@katsi Do you have any cycles to briefly go over how you got webprotege running in a kubernetes cluster?  How did you get around the initial admin account creation?

**Comment by katsi** [comment]
I did not even attempt to automate the initial admin account creation, but that to me is not a problem, because it only needs to be done once.

**Comment by mmanela** [comment]
I documented my steps here: https://matthewmanela.com/blog/deploying-webprotege-on-azure-kubernetes-service-aks/

**Comment by katsi** [comment]
Thanks for the write up! I am using our company's own CDP and Kubernetes system, so sharing it is not as helpful. Your example is overall the same that we did!

**Comment by mikeandescavage** [comment]
Thanks for the responses.  Great writeup @mmanela and that helped greatly!  Would love to contribute to allow automation for passing of the initial account creation from keyvault.  Perhaps I can lend a hand there.
